### PR TITLE
Create F04

### PR DESCRIPTION
**Pull request by krislars**
F99 is now the original from that paper, while F99FM07 is the same fundamental model with updated correlations for the NIR Rv-dependence and UV parameters from FM07. 

Future work?  FM07 model with new C5 and reduced Rv-dependence in the visible.

### CODE REVIEW

File: dust_extinction/dust_extinction.py [hunk 1]
```diff
@@ -1200,22 +1198,24 @@ def evaluate(self, in_x, Rv):
 
         # terms depending on Rv
         C2 = -0.824 + 4.717/Rv
-        # updated for FM07 correlation between C1 and C2
+        # original F99 C1-C2 correlation
         C1 = 2.030 - 3.007*C2
```
Comment (karllark) [inline dust_extinction/dust_extinction.py]: This did not change from the previous.  So was the previous really the F99 version or the FM07 version?

**Comment by karllark** [comment]
I see now what you want to do with the F99FM07.  I am wondering if having F99, F99FM07, and FM07 will be a bit confusing.  How about have the F99 the original and the FM07 the one with all the improvements?  Then we can advise people to use FM07 instead of F99.  What do you think?

**Comment by karllark** [comment]
BTW, the F99 model in this pull request is the same as before.  Was this intended?  I know this as the automated travis tests are successful.  

There is some issues with the Mac version of the tests and the building of the docs.  Trying them again to see if it was an intermittent issue with the travis computers (it happens).

File: dust_extinction/dust_extinction.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from astropy.modeling import (Model, Fittable1DModel,
                               Parameter, InputParameterError)
 
-__all__ = ['CCM89', 'FM90', 'P92', 'O94', 'F99',
+__all__ = ['CCM89', 'FM90', 'P92', 'O94', 'F99', 'F99FM07'
```
Comment (karllark) [inline dust_extinction/dust_extinction.py]: There is a comma missing after 'F99FM07' in this line.  Causing errors with the build of the docs.

**Comment by karllark** [comment]
BTW, the F99 model in this pull request is the same as before.  Was this intended?  I know this as the automated travis tests are successful.  

The build of the docs is not working.  Missing a comma in the file.  I've added a comment on the line.

**Comment by karllark** [comment]
Note that tests will be needed for the new models before we merge.  See 'test_f99.py' in the tests subdirectory for an example.

**Comment by karllark** [comment]
One other thought.  Is FM07 a new shape fitting model or a new R(V) dependent model?  Or both?  See http://dust-extinction.readthedocs.io/en/latest/dust_extinction/model_flavors.html.

File: dust_extinction/dust_extinction.py (hunk 1 above)
Comment (krislars) [reply dust_extinction/dust_extinction.py]: It is the original F99.  The newer FM07 is C1 = 2.09 - 2.84 C2.

It does gets confusing.  Here is a cheat sheet I've been making for myself to keep it straight:

image

**Comment by krislars** [comment]
Full disclosure: I just changed the one file.  Tests?  Docs?  Sounds important.  This is a good learning opportunity for me.  Let's talk.

**Comment by krislars** [comment]
Regarding F99 and FM07:  I do think that it is important to leave F99 in the code and true to the original because the stellar spectra folks use it a lot as FM_UNRED in IDL.  They will want the comparison.

The "Final Notes" of FM07 is worth reading:

> Having painted a negative picture of the relationship between R(V) and extinction properties at other wavelengths, it would be disingenuous of us to present yet another set of R(V)-dependent curves. The results of Cardelli et al. (1989), F04 (which supersede those of F99), and Valencic et al. (2004) are all reasonable, and the differences among them are instructive of the biases introduced by sample selection and methodology. It should always be remembered that these curves represent very general trends in Galactic extinction and do not constitute a standard of normalcy. Finally, although the prospect of accurately dereddening an unknown SED using a mean extinction curve is poor, this same variability demonstrates that extinction curves are responsive to local conditions, so that each one contains potentially unique information about the grains along the sight line.

I note two things here.  First, the they say that the F04 results supersede F99.  So in that sense, I would suggest that people use FM07.  On the other hand, they go on to say that this is not "yet another" set of R(V)-dependent curves.  So I'm hesitant to then present the astropy community with "yet another" model.  But as a fan of the NIR, I do really want people to recognize that the F99 is out of date.

File: dust_extinction/dust_extinction.py [hunk 3]
```diff
@@ -1228,6 +1228,136 @@ def evaluate(self, in_x, Rv):
                                  optnir_axav_x, optnir_axebv_y/Rv,
                                  self.x_range, 'F99')
 
+class F99FM07(BaseExtRvModel):
+    """
+    F99 extinction model calculation
+
+    Updated with the NIR Rv dependence in
```
Comment (karllark) [inline dust_extinction/dust_extinction.py]: Ok.  I see this info now.  Maybe just call this FM07 or something?  F99FM07 does not capture all the info (e.g., FM04).  Is everything in FM07 in this code?  If so, maybe the name could be FM07 and the documentation explain the details?

**Comment by karllark** [comment]
Looking good.  Just need to solve the issue with the failing tests now.

**Comment by krislars** [comment]
Ooh!  A math error!  I'm getting close!! :)

**Comment by krislars** [comment]
Hooray!  It passed!

**Pull request merged**